### ISSUE TITLE
wrong assertion statement for invalid slack weights (always zero) when using "p_nom" or "p_nom_opt"?

### ISSUE THREAD

**Issue by jankaeh** [PyPSA/PyPSA#711]
## Checklist

- [ ] I am using the current [`master`](https://github.com/pypsa/pypsa/tree/master) branch or the latest [release](https://github.com/pypsa/pypsa/releases).

## Describe the Bug

When using "p_nom" or "p_nom_opt" as slack_weights for pf(distribute_slack=True) the assertion that shall catch if all slack weights are zero is False even if the respective column contains values != 0:

https://github.com/PyPSA/PyPSA/blob/master/pypsa/pf.py#L395
```
not all(network.generators[slack_weights]) == 0
```

i think the simple fix is:

```
not all(network.generators[slack_weights] == 0)
```

**Comment by fneum** [comment]
Thanks, fixed via c8e9afc

**Comment by fneum** [comment]
right, fixed via 38b710c
